Services/Catalog/Catalog.UnitTests/Catalog.UnitTest/ProductControllerTests.cs

```diff
@@ -0,0 +1,229 @@
+
+            ProductService productService = new(mock.Object, mapper);
+            ProductController productController = new(mapper, productService);
+            var actionResult = await productController.Get(skip, take, new());
+
+            //Assert
+            Assert.IsType<OkObjectResult>(actionResult);
+            var actualProducts = Assert.IsAssignableFrom<List<ProductViewModel>>((actionResult as OkObjectResult).Value);
+            Assert.NotEmpty(actualProducts);
+            Assert.Equal(expectedLength, actualProducts.Count);
+            mock.Verify(unit => unit.ProductRepository, Times.Once());
+        }
+
+        [Fact]
+        public async Task Get_filteredByBrand_products_success()
+        {
+            //Arrange
+            var expectedLength = 2;
+
+            //Act
+            CatalogContext catalogContext = new(_dbOptions);
+            ProductRepository productRepository = new(catalogContext);
+            Mapper mapper = new(_mapperConfiguration);
+
+            var mock = new Mock<IUnitOfWork>();
+            mock.Setup(unit => unit.ProductRepository).Returns(productRepository);
+
+            ProductService productService = new(mock.Object, mapper);
+            ProductController productController = new(mapper, productService);
+            var actionResult = await productController.Get(0, 0, new() { Brands = new[] { _brandA } });
+
+            //Assert
+            Assert.IsType<OkObjectResult>(actionResult);
+            var actualProducts = Assert.IsAssignableFrom<List<ProductViewModel>>((actionResult as OkObjectResult).Value);
+            Assert.NotEmpty(actualProducts);
+            Assert.Equal(expectedLength, actualProducts.Count);
+            mock.Verify(unit => unit.ProductRepository, Times.Once());
+        }
+
+        [Fact]
+        public async Task Get_products_filteredByType_success()
+        {
+            //Arrange
+            var expectedLength = 2;
+
+            //Act
+            CatalogContext catalogContext = new(_dbOptions);
+            ProductRepository productRepository = new(catalogContext);
+            Mapper mapper = new(_mapperConfiguration);
+
+            var mock = new Mock<IUnitOfWork>();
+            mock.Setup(unit => unit.ProductRepository).Returns(productRepository);
+
+            ProductService productService = new(mock.Object, mapper);
+            ProductController productController = new(mapper, productService);
+            var actionResult = await productController.Get(0, 0, new() { Types = new[] { _typeA } });
+
+            //Assert
+            Assert.IsType<OkObjectResult>(actionResult);
+            var actualProducts = Assert.IsAssignableFrom<List<ProductViewModel>>((actionResult as OkObjectResult).Value);
+            Assert.NotEmpty(actualProducts);
+            Assert.Equal(expectedLength, actualProducts.Count);
+            mock.Verify(unit => unit.ProductRepository, Times.Once());
+        }
+
+
+
+        private List<Product> GetFakeProducts()
```

Comment:
Better store fakes as separate class. May be useful if you have multiple test classes